### ISSUE TITLE
Missing "orange" warning while reaching the tile size limit

### ISSUE THREAD

**Issue by kmandov** [mapbox/mapbox-studio-classic#1367]
Hi,

I am uploading a custom data source to Mapbox.

According to Mapbox Studio the tiles are within the 500K limit for zoom levels(11-16). The orange warning is missing:
![no-warning](https://cloud.githubusercontent.com/assets/2805297/7931605/aa2cefd4-091a-11e5-8162-a33ded0746f2.PNG)

...but when I actually upload the data source, there's an error:
![limit-exceeeded](https://cloud.githubusercontent.com/assets/2805297/7931619/c011a722-091a-11e5-9114-9ec2163d9ffc.PNG)

It turns out that I need to increase the min level up to 14, to meet the size limits.

I thought I should see the "orange" warning for zoom levels bellow 14. Am I missing something? 

Regards,
Kiril

**Comment by wilhelmberg** [comment]
@kmandov 
It seems, the size display is a mean value of all the currently visible tiles.
Pan around and see how the values change.

I assume you are using some kind of file (shape, geojson, kml, ...) as your custom data source.
These having the disadvantage that a lot of data gets stuffed into a single tile at lower zoom levels, as all data is always available.
At zoom 11 these footprints cannot even be distinguished anymore and it doesn't make sense to carry around all this data.

Best solution would be to move your data into PostGIS and simplify the geometries at lower zoom levels to reduce size per tile.

A guide on how to do this can be found here:
https://www.mapbox.com/guides/postgis-manual/
https://github.com/mapbox/postgis-vt-util

**Comment by wilhelmberg** [comment]
refs #1317 

I think the tile size display is broken.

/cc @camilleanne maybe here: https://github.com/mapbox/mapbox-studio/commit/639c0cb9575e5d1c62da25474bc1726339705f79

---

13MB GeoJSON with ~14,000 features.

**Zoom level 14**, I expect to see bigger tile sizes and warning:

![image](https://cloud.githubusercontent.com/assets/4549888/8003136/240870d8-0b75-11e5-8abe-5eba12114691.png)

**Zoom level 22**, I don't expect to see tile sizes that big and a warning:

![image](https://cloud.githubusercontent.com/assets/4549888/8003138/29784688-0b75-11e5-9a90-67e3dfd892a5.png)

**Comment by kmandov** [comment]
Thanks @BergWerkGIS 

I am indeed using a shapefile and it is true that at smaller zoom levels there is a lot of data stuffed into single tiles.

What I find strange is that I don't see a warning from Mapbox Studio about the reached limit. Is that a bug or I am missing something?

**Comment by camilleanne** [comment]
@BergWerkGIS looks like the orange boxes/calculation of tile size is actually handled here: https://github.com/mapbox/mapbox-studio/blob/mb-pages/app/lib.js#L41-L122

investigating -- not sure if anything's changed recently though.

**Comment by camilleanne** [comment]
So I did some fiddling with an extract of OSM and haven't been able to reproduce this. Could either @BergWerkGIS or @kmandov share with me your datasource that's seen this issue? The berlin extract I was using has very defined bounds, I'm wondering if data sets that are more organic in bounds/shape but still dense are more prone to this.

What I did find is that the calculation requires requires tiles to load fully to get the size, and if some tiles load, but other ones loaded with data are slow and you move between zoom levels or pan the map, those tiles won't make it into the calculation

![tilewarning](https://cloud.githubusercontent.com/assets/3952537/8265453/bcb0eea2-16ae-11e5-8900-f374ddcd5d7e.gif)

**Comment by wilhelmberg** [comment]
@camilleanne sure, I used the one referenced here:
https://github.com/mapnik/mapnik/issues/2873#issuecomment-106766945

**Comment by kmandov** [comment]
@camilleanne Yes. I have used the NYC Building Footprints. You can get the shapefile here: https://data.cityofnewyork.us/download/tb92-6tj8/application/zip
